packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1620,11 +1620,15 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
   @override
   TextEditingValue get currentTextEditingValue => _value;
 
+  bool _updateEditingValueInProgress = false;
```

Comment:
This PR really reminds me of this issue, which also deals with the asynchronous communication between the engine and framework: https://github.com/flutter/flutter/issues/61282

My solution there wasn't anything like this PR, but I think using this flag `_updateEditingValueInProgress` could maybe help us handle problems like that.